### PR TITLE
[std cleanup] nuke "using namespace std" under xbmc/

### PR DESCRIPTION
**Pull request by hudokkow**
This is the first PR in a series to clean up std crap.
I'll squash after review.

### CODE REVIEW

File: xbmc/PartyModeManager.cpp [hunk 1]
```diff
@@ -685,12 +684,12 @@ void CPartyModeManager::AddToHistory(int type, int songID)
...
-  m_history.push_back(make_pair(type,songID));
+  m_history.push_back(std::make_pair(type,songID));
 }
 
-void CPartyModeManager::GetRandomSelection(vector< pair<int,int> >& in, unsigned int number, vector< pair<int,int> >& out)
+void CPartyModeManager::GetRandomSelection(std::vector< std::pair<int,int> >& in, unsigned int number, std::vector< std::pair<int,int> >& out)
 {
-  number = min(number, (unsigned int)in.size());
+  number = std::min(number, (unsigned int)in.size());
   random_shuffle(in.begin(), in.end());
```
Comment (Paxxi) [inline xbmc/PartyModeManager.cpp]: Missed this one

**Comment by Paxxi** [comment]
I like it, however I think we should fix missing includes as well if we're going to touch most files. Many standard headers are missing all over the place like utility for pair, algorithm for random_shuffle.

**Comment by Paxxi** [comment]
Judging by the pattern of missed std:: functions I assume this was a find and replace job. I would suggest starting with some of our smaller source files and work your way up from there, it's hard to catch everything without manual inspection.

**Comment by hudokkow** [comment]
Yeah, sorry about the noise @Paxxi

Eclipse is acting up on me with the freaking scalability mode and unresolved stuff.
Rebased on current master, compiled and runtime tested under Ubuntu.

**Comment by Paxxi** [comment]
jenkins build this please

**Comment by hudokkow** [comment]
Errors seem unrelated. jenkins going nuts again?

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by Paxxi** [comment]
I've yet to have a chance to go through all of this, you happened to pick some of our largest source files to start with so it's taking some time.

**Comment by hudokkow** [comment]
No problem, @Paxxi 
I'm know it's complicated to poke a hornet's nest... ;-)

**Comment by Paxxi** [comment]
I think this is good to go in after a rebase, does anyone have any objections to this?

**Comment by FernetMenta** [comment]
IMO those kind of changes should not go in during beta.

**Comment by mkortstiege** [comment]
Now that we basically only have 1-2 month to drop in larger changes it has to go in during beta or it will rot on github forever. My 2c.

**Comment by Paxxi** [comment]
@FernetMenta There should be no change in functionality by this so it's very low risk to merge during this window. With that said this doesn't change any functionality so it's not a high priority item either

**Comment by hudokkow** [comment]
I can't rebase today or in the next few days. Might as well wait for J*

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by hudokkow** [comment]
It won't build. https://github.com/hudokkow/xbmc/commit/93a2f9feda6df6baca392872ab295a516e505c02 is WIP

**Comment by hudokkow** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
nudging for whenever you have time for rebase and some final review  :)

**Comment by hudokkow** [comment]
Closing this one. I'm going to PR by folder. Rebase is a nightmare now.

**Pull request closed without merging**